firmware/boron-ins-fsm/src/stateMachine.cpp

```diff
@@ -407,7 +409,10 @@ void getHeartbeat(){
       writer.endObject(); // end heartbeat message
       Particle.publish("Heartbeat", heartbeatMessage, PRIVATE);
       Log.warn(heartbeatMessage);
-      lastHeartbeatPublish = millis();
+      if ((millis()-lastHeartbeatPublish) > SM_HEARTBEAT_INTERVAL || lastHeartbeatPublish == 0){
+        lastHeartbeatPublish = millis();
+      }
```

Comment:
Do you still need this `if`-statement? Or could you go back to just always updating `lastHeartbeatPublish`?

Reply:
Again it depends on what we want to do. Right now lets say the boron connects to the console at 1pm. We would get a heartbeat every 10min. If i were to have not received a door message for 15min and then someone opens the door i would receive an extra heartbeat but the interval of 10 mins would remain the same.
For example
Start up heartbeat: 1:00
Regular heartbeat:  1:10
Regular heartbeat:  1:20
IM21 heartbeat:       1:24 (because we haven't received a IM21 broadcast for 15mins)
Regular heartbeat:  1:30
Regular heartbeat:  1:40

Removing it would result in changing the 10min interval and reset it
For example
Start up heartbeat: 1:00
Regular heartbeat:  1:10
Regular heartbeat:  1:20
IM21 heartbeat:       1:24 (because we haven't received a IM21 broadcast for 15mins)
Regular heartbeat:  1:34
Regular heartbeat:  1:44

I can see the case being made for both sides. What do you think?